StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Controllers/PayWithLinkViewController-WalletViewController.swift

```diff
@@ -645,37 +660,69 @@ extension PayWithLinkViewController.WalletViewController: LinkPaymentMethodPicke
     }
 
     func paymentDetailsPickerDidTapOnAddPayment(_ pickerView: LinkPaymentMethodPicker) {
-        if context.elementsSession.onlySupportsLinkBank {
-            // If this business is bank-only, bypass the new payment method flow and go straight to connections
-            confirmButton.update(state: .processing)
-            pickerView.setAddPaymentMethodButtonEnabled(false)
-            coordinator?.startInstantDebits { [weak self] result in
-                guard let self = self else { return }
-                switch result {
-                case .success(let paymentDetails):
-                    self.didUpdate(paymentMethod: paymentDetails, confirmationExtras: nil)
-                case .failure(let error):
-                    switch error {
-                    case InstantDebitsOnlyAuthenticationSessionManager.Error.canceled:
-                        break
-                    default:
-                        self.updateErrorLabel(for: error)
-                    }
-                }
-                self.paymentPicker.setAddPaymentMethodButtonEnabled(true)
-                self.updateUI(animated: false)
+        let supportedPaymentDetailsTypes = linkAccount.supportedPaymentDetailsTypes(for: context.elementsSession)
+
+        let bankAndCard = [ConsumerPaymentDetails.DetailsType.bankAccount, .card]
+        if bankAndCard.allSatisfy(supportedPaymentDetailsTypes.contains) {
+            let alertController = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+
+            let addBankAction = UIAlertAction(
+                title: STPLocalizedString(
+                    "Bank",
+                    "Label shown in the payment type picker describing a bank payment"
+                ),
+                style: .default
+            ) { [weak self] _ in
+                self?.addBankAccount()
             }
+            alertController.addAction(addBankAction)
```

Comment:
We'll need to check if the account supports bank accounts. Web has some logic [here](https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/consumer/link_app/frontend/src/contexts/ConsumerAccountContext/FinancialConnections/flags.ts) for that, but we could (even should!) see if we can move it to the backend.